snmplib/snmpusm.c

```diff
@@ -4339,8 +4384,24 @@ usm_read_user(const char *line)
         user->privKeyLen = proper_length;
     }
 
+    if (line == NULL) {
+        DEBUGMSGTL(("usm", "Configuration line missing public string\n"));
+        usm_free_user(user);
+        return NULL;
+    }
     line = read_config_read_octet_string(line, &user->userPublicString,
                                          &user->userPublicStringLen);
+
+    /*
+     * set the lcd entry for this engineID to the minimum boots/time
+     * values so that its a known engineid and won't return a report pdu.
+     * This is mostly important when receiving v3 traps so that the usm
+     * will at least continue processing them.
+     * Note: We do this at the end so that it only runs if the parsing
+     * was successful
+     */
+    set_enginetime(user->engineID, user->engineIDLen, 1, 0, 0);
```

Comment:
Why to move this call to the end? I think this involves a behavior change and I'm not sure that behavior change is correct. How about preserving all the if (line == NULL) checks and moving the set_enginetime() call back to its original location?

Reply:
I moved the call to the end because I didn't want to call `set_enginetime` and set it to an erroneous value if the line ends up being discarded due to a later required configuration value not being present. If you think it should be moved back up, even if the call to `usm_read_user` returns NULL, please confirm, and I will do so.